lib/checksum.js

```diff
@@ -16,14 +16,27 @@
 
 'use strict';
 
-const _ = require('lodash');
-const Bluebird = require('bluebird');
+const crypto = require('crypto');
 const sliceStream = require('slice-stream2');
 const progressStream = require('progress-stream');
 const CRC32Stream = require('crc32-stream');
-const DevNullStream = require('dev-null-stream');
 const ImageReadStream = require('./image-read-stream');
-const utils = require('./utils');
+const Bluebird = require('bluebird');
+
+/**
+ * @summary Get a hash stream
+ * @function
+ * @private
+ *
+ * @param {String} algorithm - either `crc32` or anything supported by `crypto.Hash`
+ * @returns {Stream.Transform}
+ */
+const getHashStream = (algorithm) => {
+  if (algorithm === 'crc32') {
+    return new CRC32Stream();
+  }
+  return crypto.createHash(algorithm);
```

Comment:
Yeah, it throws – I'm going to make it reject the returned promise instead of blowing up